lib/membership_pair_widget.dart

```diff
@@ -0,0 +1,175 @@
+import 'package:casual/cable_provider.dart';
+import 'package:casual/membership_pair_channel.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:flutter_webrtc/webrtc.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+import 'local_stream_provider.dart';
+import 'membership_pair.dart';
+import 'membership_pair_entry.dart';
+import 'web_rtc_answer.dart';
+import 'web_rtc_ice_candidate.dart';
+import 'web_rtc_offer.dart';
+
+const PEER_CONFIG = {
+  'iceServers': [
+    {'url': 'stun:stun.l.google.com:19302'},
+  ]
+};
+
+const SESSION_CONSTRAINTS = {
+  'mandatory': {
+    'OfferToReceiveAudio': true,
+    'OfferToReceiveVideo': true,
+  },
+  'optional': [],
+};
+
+class MembershipPairWidget extends HookWidget {
+  final MembershipPairEntry membershipPairEntry;
+
+  const MembershipPairWidget({ Key key, @required this.membershipPairEntry }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final cable = useProvider(cableProvider);
+
+    final membershipPairState = useState(MembershipPair.empty());
```

Comment:
I suspect a notifier pattern might simplify some of this code, especially in terms of the `on*Created` methods a few lines down, but this was accessible to me so I went with it.